File: lib/detectors/credentials.js
module.exports = {
  name: 'Hardcoded Credentials',
  severity: 'high',
  run(content) {
    const out = [];
    const re = /(?:password|passwd|pwd|secret|client_secret|api_secret)\s*[:=]\s*["']([^"']{4,200})["']/ig;
    let m;
    const seen = new Set();
    while ((m = re.exec(content)) !== null) {
      const val = m[1];
      if (seen.has(val)) continue;
      seen.add(val);
      const idx = m.index;
      const context = content.substr(Math.max(0, idx - 40), 120).replace(/\n/g, ' ');
      out.push({ value: val, context, severity: 'high' });
    }
    return out;
  }
};

File: JSX/detectors/credentials.py
"""Hardcoded credentials detector."""

import re
from .base import BaseDetector


class Detector(BaseDetector):
    """Detects hardcoded credentials in JavaScript."""

    def __init__(self):
        super().__init__("Hardcoded Credentials", severity="high")
        self.pattern = re.compile(
            r'(?:password|passwd|pwd|secret|client_secret|api_secret|token)\s*[:=]\s*["\']([^"\']{4,200})["\']',
            re.IGNORECASE
        )

    def run(self, content):
        """Find all hardcoded credentials."""
        findings = []
        seen = set()

        for match in self.pattern.finditer(content):
            value = match.group(1)
            if value in seen:
                continue
            seen.add(value)

            context = self._get_context(content, match.start())
            findings.append({
                "value": value,
                "context": context,
                "severity": "high"
            })

        return findings
